independent-projects/arc/processor/src/test/java/io/quarkus/arc/processor/InterceptorGeneratorTest.java

```diff
@@ -59,6 +67,47 @@ public void testGenerator() throws IOException {
         // TODO test generated bytecode
     }
 
+    /**
+     * Test that an AroundConstruct method can return java.lang.Object
+     */
+    @Test
+    public void aroundConstructReturningObject() throws IOException {
```

Comment:
I think it would be better to add a test to `arc/tests`. I find these asserts rather cryptic ;-)

Also these unit tests will be likely rewritten or completely removed.

Reply:
Yeah, I was kinda expecting this reaction :) I just felt like trail-blazing a new testing strategy, but you're right it's not very readable

Reply:
Then I'll wait until https://github.com/quarkusio/quarkus/pull/1784 is in, part of the test in there could be reused

Reply:
Merged.